hazelcast/src/main/java/com/hazelcast/map/impl/query/PartitionScanRunner.java

```diff
@@ -72,11 +74,15 @@ public PartitionScanRunner(MapServiceContext mapServiceContext) {
     @SuppressWarnings("unchecked")
     public Collection<QueryableEntry> run(String mapName, Predicate predicate, int partitionId) {
         PagingPredicate pagingPredicate = predicate instanceof PagingPredicate ? (PagingPredicate) predicate : null;
-        List<QueryableEntry> resultList = new LinkedList<QueryableEntry>();
-
         PartitionContainer partitionContainer = mapServiceContext.getPartitionContainer(partitionId);
         MapContainer mapContainer = mapServiceContext.getMapContainer(mapName);
-        Iterator<Record> iterator = partitionContainer.getRecordStore(mapName).loadAwareIterator(getNow(), false);
+        RecordStore recordStore = partitionContainer.getRecordStore(mapName);
```

Comment:
Isn't `partitionContainer.getExistingRecordStore(name)` more suitable for query/readonly operations?